### PR TITLE
Feature/test power combiner

### PR DESCRIPTION
**Pull request by pslocum**
Created new directory Source/RxComponents and moved LMCReceiver, LMCChannel, LMCPatchAntenna, and LMCPowerCombiner into it.  Using existing FIRHandler instances, streamlined LMCPowerCombiner and implemented it in LMCPatchSignalGenerator, LMCPlaneWaveSignalGenerator, LMCDipoleSignalGenerator.  Also implemented FIRHandler in LMCPatchSignalGenerator.

### CODE REVIEW

File: Source/RxComponents/LMCPowerCombiner.cc [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+
+  void PowerCombiner::SetSMatrixParameters(int aPowerCombiner, int aPatchesPerStrip)
+  {
+	  nPatchesPerStrip = aPatchesPerStrip;
+
+	  if (aPowerCombiner == 0) // corporate
+	  {
+		  junctionLoss = 0.;
+		  patchLoss = 0.6;
```
Comment (buzinsky) [inline Source/RxComponents/LMCPowerCombiner.cc]: fix indentation (whole file)

File: Source/Generators/LMCPatchSignalGenerator.cc [hunk 2]
```diff
@@ -628,11 +508,9 @@ namespace locust
         std::thread Kassiopeia(KassiopeiaInit, gxml_filename);     // spawn new thread
         fRunInProgress = true;
 
-        double* filterarray = GetFIRFilter(1);
-        unsigned nfilterbins = GetNFilterBins(filterarray);
+        int nfilterbins = fReceiverFIRHandler.GetFilterSize();
+        double dtfilter = fReceiverFIRHandler.GetFilterResolution();
```
Comment (buzinsky) [inline Source/Generators/LMCPatchSignalGenerator.cc]: camel case

File: Source/Generators/LMCPlaneWaveSignalGenerator.cc [hunk 3]
```diff
@@ -28,6 +28,7 @@ namespace locust
   PlaneWaveSignalGenerator::PlaneWaveSignalGenerator( const std::string& aName ) :
     Generator( aName ),
     fLO_Frequency( 0.),
+    fphiLO(0.),
```
Comment (buzinsky) [inline Source/Generators/LMCPlaneWaveSignalGenerator.cc]: camel case

**Comment by buzinsky** [comment]
I have another comment/ question. What was the eventual planned directory structure? I think this new one makes sense.

**Comment by pslocum** [comment]
For the directory structure, we are aiming to separate transmitter classes
from receiver classes.  LMCSignal-modifying receiving classes are presently
in Source/Generators.  Transmitter classes are in Source/Transmitters
(without Kassiopeia) or Source/Kassiopeia (with Kassiopeia).  RF utilities
are presently in Source/RxComponents and in Source/Transforms, and software
core classes are in Source/Core.  The LMCParticle class might need to be
moved from Source/Core to e.g. Source/Kassiopeia?  Any other thoughts
welcome.

File: Source/Generators/LMCPatchSignalGenerator.hh [hunk 4]
```diff
@@ -94,17 +88,16 @@ namespace locust
 
 
             bool DoGenerate( Signal* aSignal );
-            void DriveAntenna(FILE *fp, int PreEventCounter, unsigned index, Signal* aSignal, double* filterarray, unsigned nfilterbins, double dtfilter);
-            void InitializePatchArray();
+            void DriveAntenna(FILE *fp, int PreEventCounter, unsigned index, Signal* aSignal, int nfilterbins, double dtfilter);
+            bool InitializePatchArray();
             bool InitializePowerCombining();
+            FIRHandler fReceiverFIRHandler;
```
Comment (psurukuc) [inline Source/Generators/LMCPatchSignalGenerator.hh]: The FIRHandler changed a little bit where we now have a core class and derived classes, so this needs to change from `FIRHandler fReceiverFIRHandler` to `FIRReceiverHandler fReceiverFIRHandler;`.

File: Source/Generators/LMCPatchSignalGenerator.cc [hunk 5]
```diff
@@ -61,15 +58,11 @@ namespace locust
...
-        if( aParam.has( "filter-filename" ) )
-        {
-            gfilter_filename = aParam["filter-filename"]().as_string();
-        }
 
-        if( aParam.has( "filter-resolution" ) )
-        {
-            fFilter_resolution = aParam["filter-resolution"]().as_double();
-        }
+       if(!fReceiverFIRHandler.Configure(aParam,false))
```
Comment (psurukuc) [inline Source/Generators/LMCPatchSignalGenerator.cc]: With the `FIRHandler` now changed to `FIRReceiverHandler`, there is no need to input false as a second argument to the Configure function in FIRHandler.

**Pull request merged**